src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -71,6 +69,12 @@ protected SecretMasker(SecretMasker copy)
         try
         {
             copy.SyncObject.EnterReadLock();
+
+            _regexEngine = copy._regexEngine;
+            _highPerformanceScanner = copy._highPerformanceScanner;
+            _highPerformanceSignatureToPatternsMap = copy._highPerformanceSignatureToPatternsMap;
+            _generateCorrelatingIds = copy._generateCorrelatingIds;
```

Comment:
regexEngine and generate correlating IDs were missing, add test coverage.

Reply:
Still working on tests here, just fixed more bugs in it.

I wonder if we really need Clone. We inherited it from the VSO interface, but I'm not sure that pipeline agent actually uses it...

Reply:
How can we build confidence to zap this? Would be good. I don't see any MS docs for this type. I very much suspect we won't have serious compatibility issues simplifying it. We could try to go through some process of marking it obsolete, etc. Better, perhaps we could agree to a scream test, first simplify accepting that we might need to quickly stub in something to restore if need be. Probably a good bet.

Reply:
Sigh, I may have spoken and acted too soon here. Clone does appear to be used in agent (in a questionable but hard to change place) :(

Reply:
I looked at things in more depth and I believe there is still a path forward to keep Clone out. Let's leave it zapped in this PR with a small possiblity that I will have to revert that.